pkg/autoscaler/autoscaler.go

```diff
@@ -457,6 +461,75 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
 				cleanupObjects[asr.GetName()] = asr
 			}
 
+			for _, machineSet := range transientMachineSets {
+				By(fmt.Sprintf("Waiting for MachineSet %s to acquire autoscaling annotations", machineSet.GetName()))
+				Eventually(komega.Object(machineSet), framework.WaitShort, pollingInterval).Should(HaveField("GetAnnotations()", SatisfyAll(
+					HaveKey(caMinSizeAnnotation),
+					HaveKey(caMaxSizeAnnotation),
+				)), "MachineSet %s did not acquire autoscaling annotation", machineSet.GetName())
+			}
+
+			// on some cloud providers, we have experienced the new nodes having resources added to their status.capacity
+			// after they have been initialized. this has been seen with `hugepages-1Gi` and `hugepages-2Mi`. so we
+			// wait until the nodes have both entries before moving on.
+			By("Waiting for the new Nodes to have hugepages-1Gi and hugepages-2Mi capacity")
+			Eventually(func() (bool, error) {
+				nodes := []*corev1.Node{}
+
+				for _, machineSet := range transientMachineSets {
+					if n, err := framework.GetNodesFromMachineSet(client, machineSet); err != nil {
+						return false, err
+					} else if len(n) != 1 {
+						return false, fmt.Errorf("expected 1 node for MachineSet %s, found %d", machineSet.GetName(), len(n))
+					} else {
+						nodes = append(nodes, n[0])
+					}
+				}
+
+				for _, n := range nodes {
+					_, ok1Gi := n.Status.Capacity[corev1.ResourceHugePagesPrefix+"1Gi"]
+					_, ok2Mi := n.Status.Capacity[corev1.ResourceHugePagesPrefix+"2Mi"]
+					if !ok1Gi || !ok2Mi {
+						return false, nil
+					}
+				}
+
+				return true, nil
+			}, framework.WaitMedium, pollingInterval).Should(BeTrue(), "Node capacity resources did not contain hugepages-1Gi and hugepages-2Mi")
+
+			// wait until the new nodes have the same resource keys before progressing, otherwise the balance will not work.
+			By("Waiting for the new Nodes to have similar resources")
+			Eventually(func() (bool, error) {
+				nodes := []*corev1.Node{}
+
+				for _, machineSet := range transientMachineSets {
+					if n, err := framework.GetNodesFromMachineSet(client, machineSet); err != nil {
+						return false, err
+					} else if len(n) != 1 {
+						return false, fmt.Errorf("expected 1 node for MachineSet %s, found %d", machineSet.GetName(), len(n))
+					} else {
+						nodes = append(nodes, n[0])
+					}
+				}
+
+				// add the resources from each node's capacity, counting the number of each
+				resources := map[corev1.ResourceName]int{}
+				for _, n := range nodes {
+					for k := range n.Status.Capacity {
+						resources[k] += 1
+					}
+				}
+
+				// if any of the entries does not equal 2, it means that only one of the nodes had that resource
+				for _, v := range resources {
+					if v != 2 {
+						return false, nil
+					}
+				}
+
+				return true, nil
+			}, framework.WaitMedium, pollingInterval).Should(BeTrue(), "Node capacity resources did not match")
```

Comment:
We could rewrite this a little to do the final matching using gomega, in that case you'd be able to see which resources didn't match, might help out a little?
```suggestion
			Eventually(func() (map[string]int, error) {
				nodes := []*corev1.Node{}

				for _, machineSet := range transientMachineSets {
					if n, err := framework.GetNodesFromMachineSet(client, machineSet); err != nil {
						return false, err
					} else if len(n) != 1 {
						return false, fmt.Errorf("expected 1 node for MachineSet %s, found %d", machineSet.GetName(), len(n))
					} else {
						nodes = append(nodes, n[0])
					}
				}

				// add the resources from each node's capacity, counting the number of each
				resources := map[corev1.ResourceName]int{}
				for _, n := range nodes {
					for k := range n.Status.Capacity {
						resources[k] += 1
					}
				}

				// Check that each resource in the list has a count of two, if any of the entries does not equal 2, it means that only one of the nodes had that resource
				return resources, nil
			}, framework.WaitMedium, pollingInterval).Should(HaveEach(2), "Node capacity resources did not match")
```
The `HaveEach` matcher will ignore the keys in the map but check that each value equals `2`.

Eg of a failure
```
        Node capacity resources did not match
        Expected
            <map[string]int | len:3>: {"a": 1, "b": 2, "c": 2}
        to contain element matching
            <int>: 2
```